src/experiments.ts

```diff
@@ -0,0 +1,172 @@
+import { bold } from "colorette";
+import * as leven from "leven";
+
+import { configstore } from "./configstore";
+import { FirebaseError } from "./error";
+
+export interface Experiment {
+  shortDescription: string;
+  fullDescription?: string;
+  public?: boolean;
+  docsUri?: string;
+  default?: boolean;
+}
+
+// Utility method to ensure there are no typos in defining ALL_EXPERIMENTS
+function experiments<Keys extends string>(exp: Record<Keys, Experiment>): Record<Keys, Experiment> {
```

Comment:
I'm confused to what this is doing. Wouldn't just defining `ALL_EXPERIMENTS: Record<string, Experiments>` below enforce the same thing?

Reply:
If I called this a `Record<string, Experiments>` then the keys would be of type `string` and I can't use `keyof ALL_EXPERIMENTS` to make a typedef of experiment names. We need a literal, to preserve key names, but then if I don't type it, you neither get code complete for an experiment nor protection against mistyping the experiment. Using this utility makes the compiler enforce the type of values but infer the type of keys.